targets/javascript/make.js

```diff
@@ -97,12 +106,21 @@ function hasResultActions(apiCall) {
 function getResultActions(tabbing, apiCall) {
     if (apiCall.result === "LoginResult")
         return tabbing + "if (result != null) {\n"
-            + tabbing + "       if(result.data.SessionTicket != null) {\n"
-            + tabbing + "           PlayFab._internalSettings.sessionTicket = result.data.SessionTicket;\n"
-            + tabbing + "       }\n"
-            + tabbing + "       if (result.data.EntityToken != null) {\n"
-            + tabbing + "           PlayFab._internalSettings.entityToken = result.data.EntityToken.EntityToken;\n"
-            + tabbing + "       }\n"
+            + tabbing + "   // Capture the playFabId, and make sure the authenticationContext exists with the ID in question\n"
+            + tabbing + "   var playFabId = result.data.PlayFabId;\n"
+            + tabbing + "\n"
+            + tabbing + "   if(PlayFab._internalSettings.authenticationContext.hasOwnProperty(playFabId) == false) {\n"
+            + tabbing + "       PlayFab._internalSettings.authenticationContext[playFabId] = {};"
```

Comment:
Ok, I figured out what I don't like about this. It took me a while.
It is not the responsibility of the PlayFab SDK to archive or index the authenticationContexts for every logged in player.

PlayFab._internalSettings.authenticationContext should not be a dictionary of all authenticationContext's ever.

PlayFabClientApi.LoginWithCustomId will return an authenticationContext as part of a new parameter.
If the customer keeps it, then great. If they don't, then it's just gone. We're not tracking them.

PlayFab._internalSettings.authenticationContext is just one instance of a single authenticationContext, which is the last person to use the static interface to log in. That's it. It's over-written when somebody else logs in.

The developer is responsible for calling login 3 times, and keeping track of three separate authenticationContext objects, and feeding them back into request objects.

Reply:
Thanks Paul! Yes, that is a good description of the goal. We don't want to cache authentication contexts ourselves. It's up to customers to keep/maintain/store them if they want, but we do not keep these authentication objects in _internalSettings or anywhere else ourselves. 

---
In reply to: [257078270](https://github.com/PlayFab/SDKGenerator/pull/403#discussion_r257078270) [](ancestors = 257078270)